Sources/GRPC/ConnectionPool/ConnectionPool.swift

```diff
@@ -716,6 +747,15 @@ extension ConnectionPool: ConnectionManagerConnectivityDelegate {
     if let droppedReservations = self._connections[id]?.unavailable(), droppedReservations > 0 {
       self.streamLender.returnStreams(droppedReservations, to: self)
     }
+
+    // Now that a connection has become unavailable, we must make sure the minimum
+    // number of connections is still met.
+    if let connectionManager = self._connections[id]?.manager,
+      case .neverGoIdle = connectionManager.idleBehavior,
+      connectionManager.sync.isTransientFailure
+    {
+      connectionManager.sync.startConnecting()
+    }
```

Comment:
I'm not so sure about this. If the connection drops and is in transient failure then it will start attempting to connect after a backoff period. We shouldn't ignore that period other we risk hammering a server with attempts to connect.